sal/tests/server/unit/test_server.py

```diff
@@ -0,0 +1,140 @@
+                                         None,
+                                         token_secret,
+                                         token_lifetime)
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture],
+          max_examples=10)
+@given(token_secret=st.text(max_size=AUTH_TOKEN_SECRET_MIN_LENGTH - 1),
+       token_lifetime=token_lifetime)
+def test_authenticator_validation_short_key(token_secret,
+                                            token_lifetime,
+                                            server_with_auth_prov,
+                                            mock_authentication_provider):
+
+    """
+    Tests that a server authentication is invalid if the key length is less
+    than the minimum
+
+    GIVEN
+        A server with an authenticator
+        AND a secret key shorter than the minimum length
+        AND a non-zero token lifetime
+    WHEN
+        The server attemped to validate the authentication
+    THEN
+        The authentication is invalid (ValueError)
+    """
+
+    with pytest.raises(ValueError):
+        server_with_auth_prov._validate_auth(mock_authentication_provider,
+                                             None,
+                                             token_secret,
+                                             token_lifetime)
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture],
+          max_examples=10)
+@given(token_secret=token_secret,
+       token_lifetime=st.floats(max_value=0))
+def test_authenticator_validation_invalid_token_lifetime(
+    token_secret,
+    token_lifetime,
+    server_with_auth_prov,
+    mock_authentication_provider):
+
+    """
+    Tests that the server authentication is invalid without a non-zero token
+    lifetime
+
+    GIVEN
+        A server with an authenticator
+        AND a secret key longer than the minimum length
+        AND a zero or negative token lifetime
+    WHEN
+        The server attemped to validate the authentication
+    THEN
+        The authentication is invalid (ValueError)
+    """
+
+    with pytest.raises(ValueError):
+        server_with_auth_prov._validate_auth(mock_authentication_provider,
+                                             None,
+                                             token_secret,
+                                             token_lifetime)
+
+
+@settings(suppress_health_check=[HealthCheck.function_scoped_fixture],
+          max_examples=10)
+@given(token_secret=token_secret,
+       token_lifetime=token_lifetime)
+def test_authenticator_validation_authorisation_without_authentication(
+    token_secret,
+    token_lifetime,
+    server):
+
+    """
+    Tests that the server authentication is invalid if it will authorise but
+    not authenticate
+
+    GIVEN
+        A server without an authenticator
+        AND the server has an authoriser
+        AND a secret key longer than the minimum length
+        AND a zero or negative token lifetime
```

Comment:
**issue:** As in some other tests, this appears to be a copy/paste error.
```suggestion
        AND a non-zero token lifetime
```

Reply:
Thanks - I clearly should have cast my eye over the docstrings again!